posydon/popsyn/synthetic_population.py

```diff
@@ -239,22 +288,41 @@ def parse(self, S1_state=None, S2_state=None, binary_state=None,
                 if any(sel):
                     df_tmp = pd.DataFrame()
                     df_tmp = df.loc[sel]
-                    sel_met.extend(sel)
                     # store metallicity
                     df_tmp['metallicity'] = met
                     # concatenate results
                     df_sel_met = pd.concat([df_sel_met, df_tmp])
                     del df_tmp
 
+            # check last binary if it should be included
+            if last_binary_df is not None:
+                logic = self.apply_logic(last_binary_df, 
+                                    S1_state     = S1_state,
+                                    S2_state     = S2_state,
+                                    binary_state = binary_state,
+                                    binary_even  = binary_event,
+                                    step_name    = step_name,
+                                    invert_S1S2  = invert_S1S2)
+                
+                # The last binary is selected
+                if any(logic) == True:
+                    df_tmp = last_binary_df.loc[logic]
+                    df_sel_met = pd.concat([df_sel_met, df_tmp])
+
+            # get unique indicies
+            sel_met = df_sel_met.index.unique()
```

Comment:
Could you please tell, why you use `unique` instead of `drop_duplicates` here, cf. `sel = df.loc[logic].index.drop_duplicates()` being used before?

Reply:
`unique` provides a numpy array, while `drop_duplicates` provides a pandas Series. It shouldn't matter much in practice, except maybe some memory usage.

Reply:
If we don't care here, which to use, we should use the same for consistency.